### ISSUE TITLE
new image bbox

### ISSUE THREAD

**Issue by 21157651** [BAAI-WuDao/BriVL#1]
how to get 'bbox' in BriVL/BriVL-code-inference/data/jsonls/example.jsonl

**Comment by knaffe** [comment]
The bbox of these examples have 100 ROI. How to use FasterRCNN to detect these crazy numbers of Objects?

**Comment by SereneQI** [comment]
I used Detectron2 with weight _mask_rcnn_R_50_FPN_3x.yaml_ to get 100 candidate bboxes, but the coordinates are not exactly the same as that in the _example.jsonl_. So I'd like to know if the object detector used in the project could be provided for complete reproduction.

**Comment by chuhaojin** [comment]
BriVL uses the Bottom-Up Attention model as its object detection tool,  this model can be obtained from[BriVL-BUA-applications](https://github.com/chuhaojin/BriVL-BUA-applications)

**Comment by knaffe** [comment]
By the way,  I have test the AIC-ICC Validset from BriVL-API1.0, but the retrieval result is too low (Recall@1 < 1%). I use your released code for 'retrieval ' and Faiss vector retrieval, but still have disappointing result.  Could you release more details in this Exp of Paper?

**Comment by zgj-gutou** [comment]
> BriVL uses the Bottom-Up Attention model as its object detection tool, this model can be obtained from[BriVL-BUA-applications](https://github.com/chuhaojin/BriVL-BUA-applications)

hi, I used [BriVL-BUA-applications](https://github.com/chuhaojin/BriVL-BUA-applications)  to get the bboxes. I modified the extract-bua-caffe-r101.yaml file. The MAX_BOXES is changed from 45 to 100, but the coordinates are not exactly the same as that in the example.jsonl. So I am a little confused. I don't know what is wrong during using the [BriVL-BUA-applications](https://github.com/chuhaojin/BriVL-BUA-applications). Can you give a example from the example.jsonl and generate bboxes of it by using [BriVL-BUA-applications](https://github.com/chuhaojin/BriVL-BUA-applications). Thank you very much!

**Comment by chuhaojin** [comment]
> > BriVL uses the Bottom-Up Attention model as its object detection tool, this model can be obtained from[BriVL-BUA-applications](https://github.com/chuhaojin/BriVL-BUA-applications)
> 
> hi, I used [BriVL-BUA-applications](https://github.com/chuhaojin/BriVL-BUA-applications) to get the bboxes. I modified the extract-bua-caffe-r101.yaml file. The MAX_BOXES is changed from 45 to 100, but the coordinates are not exactly the same as that in the example.jsonl. So I am a little confused. I don't know what is wrong during using the [BriVL-BUA-applications](https://github.com/chuhaojin/BriVL-BUA-applications). Can you give a example from the example.jsonl and generate bboxes of it by using [BriVL-BUA-applications](https://github.com/chuhaojin/BriVL-BUA-applications). Thank you very much!

Due to the difference between the library versions or the machines, the results of the bounding box will be slightly random, which will not affect the performance of BriVL. In addition, you can calculate the IoU values of these two sets of bounding boxes to verify their correctness.

**Comment by chuhaojin** [comment]
We just fixed a bug: Change the image size in cfg/test.yml to 380. Please pay attention to this when using BriVL, sorry for the inconvenience.

**Comment by troilus-canva** [comment]
> > BriVL uses the Bottom-Up Attention model as its object detection tool, this model can be obtained from[BriVL-BUA-applications](https://github.com/chuhaojin/BriVL-BUA-applications)
> 
> hi, I used [BriVL-BUA-applications](https://github.com/chuhaojin/BriVL-BUA-applications) to get the bboxes. I modified the extract-bua-caffe-r101.yaml file. The MAX_BOXES is changed from 45 to 100, but the coordinates are not exactly the same as that in the example.jsonl. So I am a little confused. I don't know what is wrong during using the [BriVL-BUA-applications](https://github.com/chuhaojin/BriVL-BUA-applications). Can you give a example from the example.jsonl and generate bboxes of it by using [BriVL-BUA-applications](https://github.com/chuhaojin/BriVL-BUA-applications). Thank you very much!

I can reproduce the bboxes same as those in [example.jsonl](https://github.com/BAAI-WuDao/BriVL/blob/main/BriVL-code-inference/data/jsonls/example.jsonl)

**Comment by zgj-gutou** [comment]
> > > BriVL uses the Bottom-Up Attention model as its object detection tool, this model can be obtained from[BriVL-BUA-applications](https://github.com/chuhaojin/BriVL-BUA-applications)
> > 
> > 
> > hi, I used [BriVL-BUA-applications](https://github.com/chuhaojin/BriVL-BUA-applications) to get the bboxes. I modified the extract-bua-caffe-r101.yaml file. The MAX_BOXES is changed from 45 to 100, but the coordinates are not exactly the same as that in the example.jsonl. So I am a little confused. I don't know what is wrong during using the [BriVL-BUA-applications](https://github.com/chuhaojin/BriVL-BUA-applications). Can you give a example from the example.jsonl and generate bboxes of it by using [BriVL-BUA-applications](https://github.com/chuhaojin/BriVL-BUA-applications). Thank you very much!
> 
> I can reproduce the bboxes same as those in [example.jsonl](https://github.com/BAAI-WuDao/BriVL/blob/main/BriVL-code-inference/data/jsonls/example.jsonl)

hello, how can you do that ? Can you tell me what is changed in the extract-bua-caffe-r101.yaml file ? thank you!!!

**Comment by troilus-canva** [comment]
> > > > BriVL uses the Bottom-Up Attention model as its object detection tool, this model can be obtained from[BriVL-BUA-applications](https://github.com/chuhaojin/BriVL-BUA-applications)
> > > 
> > > 
> > > hi, I used [BriVL-BUA-applications](https://github.com/chuhaojin/BriVL-BUA-applications) to get the bboxes. I modified the extract-bua-caffe-r101.yaml file. The MAX_BOXES is changed from 45 to 100, but the coordinates are not exactly the same as that in the example.jsonl. So I am a little confused. I don't know what is wrong during using the [BriVL-BUA-applications](https://github.com/chuhaojin/BriVL-BUA-applications). Can you give a example from the example.jsonl and generate bboxes of it by using [BriVL-BUA-applications](https://github.com/chuhaojin/BriVL-BUA-applications). Thank you very much!
> > 
> > 
> > I can reproduce the bboxes same as those in [example.jsonl](https://github.com/BAAI-WuDao/BriVL/blob/main/BriVL-code-inference/data/jsonls/example.jsonl)
> 
> hello, how can you do that ? Can you tell me what is changed in the extract-bua-caffe-r101.yaml file ? thank you!!!

I didn't change anything except device from cuda to cpu, as I'm running it on mac. And run the command mentioned in the readme `python3 bbox_extractor.py --img_path ../BriVL/data/imgs/baike_14014334_0.jpg --out_path test_data/test1.npz`

**Comment by Qiulin-W** [comment]
> By the way, I have test the AIC-ICC Validset from BriVL-API1.0, but the retrieval result is too low (Recall@1 < 1%). I use your released code for 'retrieval ' and Faiss vector retrieval, but still have disappointing result. Could you release more details in this Exp of Paper?

Hi, I got a similar results as yours on the AIC-ICC validation set (30000 images with 5 captions for each image): i2t R1: 1.57%, t2i R1:0.48%
After going into details, I found the model did provide some resonable results, e.g.
![Screenshot_from_2021-10-19_17-42-47](https://user-images.githubusercontent.com/37483535/137896120-e2f2008b-9a53-4260-a604-110d14e9bc83.png)

The highlighted text in the bottom-left is the query text, the ground truth image is above the text.  The three images on the right are the top-3 images matched by the model. However, as the example shows, the model only matches words "裙子", "女孩" and ignores other information, which severely affect the recall.

Moreover, I found another paper (https://arxiv.org/abs/2109.04699v2) that did the same evaluation on AIC-ICC dataset.  In their paper, they mentioned that they conducted experiments on the "test subset" of AIC-ICC, which only contains 10000 "data". The results reported in their paper about the WenLan model are similar as those reported in WenLan paper. But the validation set contains 30000 images and 150000 captions instead.
<img width="289" alt="E-CLIP_dataset_detail" src="https://user-images.githubusercontent.com/37483535/137895880-7983b6c5-a245-45d0-b016-42dc4f4cdfa7.png">
 
May the authors @chuhaojin provide more details of the test set and potential pre-processing procedures? Many thanks!

**Comment by huang-xx** [comment]
@Qiulin-W Obviously, Efficientclip’s paper uses the data from Wenlan directly (because they are the same), but Efficientclip’s paper says that they obtained the results on the test set. Still, the data in Wenlan is obtained from the validation set (in addition,  As far as I know, the AIC-ICC test set has never been made public).

![image](https://user-images.githubusercontent.com/10807108/141401190-35455b86-bdfd-457d-b240-1b27de779ebd.png)

![image](https://user-images.githubusercontent.com/10807108/141401266-abd34ab1-c0d9-4cdb-be08-75d930cf1e6b.png)


At the same time, according to the paper of [wenlan2.0](https://arxiv.org/pdf/2110.14378), their new experiment was obtained on the first 10,000 samples in the validation set.

![image](https://user-images.githubusercontent.com/10807108/141401329-7cdba0ee-233c-41be-b228-75b7399a56f9.png)

**Comment by huang-xx** [comment]
@Qiulin-W  @knaffe @chuhaojin  The following results are tested on AIC-ICC validate dataset using the code of this repo. I can ensure that the processing results of jsonl file are exactly the same as the file provided in the example.
![image](https://user-images.githubusercontent.com/10807108/142400603-27038ec6-2d84-4a6c-8d67-1e19bdcfb694.png)

This result is far inferior to the result in the paper. Any suggestions?

**Comment by chuhaojin** [comment]
> @Qiulin-W @knaffe @chuhaojin The following results are tested on AIC-ICC validate dataset using the code of this repo. I can ensure that the processing results of jsonl file are exactly the same as the file provided in the example. ![image](https://user-images.githubusercontent.com/10807108/142400603-27038ec6-2d84-4a6c-8d67-1e19bdcfb694.png)
> 
> This result is far inferior to the result in the paper. Any suggestions?

@huang-xx @Qiulin-W Sorry, I don’t know more about the evaluation details of the BriVL model. You can consult the students in the Model Development Group(who take charge of this part) for more details. FYI: https://mp.weixin.qq.com/s/580JGzl5fUe7WnPaI3WPWQ (Refer to the bottom of this article).

**Comment by chuhaojin** [comment]
> @Qiulin-W @knaffe @chuhaojin The following results are tested on AIC-ICC validate dataset using the code of this repo. I can ensure that the processing results of jsonl file are exactly the same as the file provided in the example. ![image](https://user-images.githubusercontent.com/10807108/142400603-27038ec6-2d84-4a6c-8d67-1e19bdcfb694.png)
> 
> This result is far inferior to the result in the paper. Any suggestions?

@huang-xx @knaffe Sorry, I don’t know more about the evaluation details of the BriVL model. You can consult the student in the Model Development Group(@moonlitt, who takes charge of this part) for more details.

**Comment by huang-xx** [comment]
@chuhaojin After fine-tuning your pre-trained model on AIC-ICC train datasets，I can get similar results as your paper. 

![51F81EE0-3748-45C7-B200-A94DAFB3568A](https://user-images.githubusercontent.com/10807108/143162507-26d7e7d7-db79-4c7c-a4ed-c90d7e923398.jpg)
Is the result of the baseline CLIP model in your paper also fine-tuning?

Thank you for your open-source.

**Comment by jim4399266** [comment]
@moonlitt Hello, my evaluation results (i2t r@1 1.09% ; t2i r@1 0.37% )  on AIC-ICC validate dataset(I used 30000 samples) are also far from  the results in  paper. Could you please share the evaluation codes as a reference?
